npm/metrics/prometheus-metrics.go

```diff
@@ -24,16 +30,33 @@ var (
 	addIPSetExecTime   prometheus.Summary
 	numIPSetEntries    prometheus.Gauge
 	ipsetInventory     *prometheus.GaugeVec
+
+	policyApplyTime    *prometheus.SummaryVec
+	podApplyTime       *prometheus.SummaryVec
+	namespaceApplyTime *prometheus.SummaryVec
+
+	execTimeQuantiles = map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001}
 )
 
-// Constants for metric names and descriptions as well as exported labels for Vector metrics
+// Constants for customer cluster and node metric names and descriptions as well as exported labels for Vector metrics
 const (
 	numPoliciesName = "num_policies"
 	numPoliciesHelp = "The number of current network policies for this node"
 
 	addPolicyExecTimeName = "add_policy_exec_time"
 	addPolicyExecTimeHelp = "Execution time in milliseconds for adding a network policy"
 
+	// TODO do update/delete
+	policyApplyTimeName = "policy_apply_time"
+	policyApplyTimeHelp = "Execution time in milliseconds for updating/deleting a network policy. NOTE: for apply time for adding, see add_policy_exec_time"
+	applyModeLabel      = "apply_mode"
+
+	podApplyTimeName = "pod_apply_time"
+	podApplyTimeHelp = "Execution time in milliseconds for adding/updating/deleting a pod"
+
+	namespaceApplyTimeName = "namespace_apply_time"
```

Comment:
Is there a reason for us to combine all resource events into the same metric ? I think there is value in keeping one metric for one type of resource event. Wdyt ?

Reply:
discussed offline that using vector metrics with labels is a good strategy